x-pack/platform/plugins/shared/fleet/server/routes/epm/handlers.ts

```diff
@@ -389,6 +392,44 @@ export const createCustomIntegrationHandler: FleetRequestHandler<
     throw error;
   }
 };
+export const updateCustomIntegrationHandler: FleetRequestHandler<
+  TypeOf<typeof CustomIntegrationRequestSchema.body>
+> = async (context, request, response) => {
+  const [coreContext] = await Promise.all([context.core, context.fleet]);
+  const esClient = coreContext.elasticsearch.client.asInternalUser;
+  const soClient = coreContext.savedObjects.client;
+
+  try {
+    const { fields } = request.body as TypeOf<typeof CustomIntegrationRequestSchema.body>;
+    const { pkgName } = request.params as unknown as TypeOf<
+      typeof CustomIntegrationRequestSchema.params
+    >;
+    const result = await updateCustomIntegration(esClient, soClient, pkgName, fields);
+
+    return response.ok({
+      body: {
+        id: pkgName,
+        result,
+      },
+    });
+  } catch (error) {
+    if (error instanceof CustomIntegrationNotFoundError) {
```

Comment:
We usually do not catch error in the handler but have a global error handler for Fleet https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/fleet/server/errors/handlers.ts#L65  and we define most of our custom errors there https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/fleet/server/errors/index.ts#L32